### ISSUE TITLE
I can't reproduce the accuracy such as 97.8% and 96.8% as the paper presented.

### ISSUE THREAD

**Issue by qzcmask** [skumra/robotic-grasping#11]
Can you release the newest code and the fine-tuned model?

I want to apply your model to detect grasps.

Thank you.

**Comment by skumra** [comment]
We will be releasing the updated code and pre-trained models soon. We were waiting for the paper review process to complete. Now that the paper has been accepted at IROS 2020, we will be releasing the final models in the release 0.2.0.

**Comment by qzcmask** [comment]
Congratulations!


I am looking forward to the project.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Sulabh Kumra"<notifications@github.com&gt;;
发送时间:&nbsp;2020年7月4日(星期六) 中午12:35
收件人:&nbsp;"skumra/robotic-grasping"<robotic-grasping@noreply.github.com&gt;;
抄送:&nbsp;"戚宗城"<923746590@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [skumra/robotic-grasping] I can't reproduce the accuracy such as 97.8% and 96.8% as the paper presented. (#11)





 
We will be releasing the updated code and pre-trained models soon. We were waiting for the paper review process to complete. Now that the paper has been accepted at IROS 2020, we will be releasing the final models in the release 0.2.0.

**Comment by qzcmask** [comment]
After I ran the script `run_offline.py`, the score of generated grasp.png is 1.77 which is not between 0 and 1.

![image](https://user-images.githubusercontent.com/28850233/87380901-0c5f7380-c5c6-11ea-9879-b9a491858dbc.png)

How can I solve this problem?

**Comment by skumra** [comment]
> Can you release the newest code and the fine-tuned model?

The latest code has been released in `release 0.2.0`. The trained models will be included in the `release 0.2.1`

> After I ran the script run_offline.py, the score of generated grasp.png is 1.77 which is not between 0 and 1.

Which dataset did you use to train the model for this? I see that you are running the script on the Jacquard dataset.

**Comment by qzcmask** [comment]
Yes, I test the model on the Jacquard dataset.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "skumra/robotic-grasping"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年7月14日(星期二) 中午1:34
收件人:&nbsp;"skumra/robotic-grasping"<robotic-grasping@noreply.github.com&gt;;
抄送:&nbsp;"戚宗城"<923746590@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [skumra/robotic-grasping] I can't reproduce the accuracy such as 97.8% and 96.8% as the paper presented. (#11)





  
Can you release the newest code and the fine-tuned model?
  
The latest code has been released in release 0.2.0. The trained models will be included in the release 0.2.1
  
After I ran the script run_offline.py, the score of generated grasp.png is 1.77 which is not between 0 and 1.
  
Which dataset did you use to train the model for this? I see that you are running the script on the Jacquard dataset.

**Comment by skumra** [comment]
You can normalize the Q output of the model if you want the output to be between 0 and 1.

**Comment by qzcmask** [comment]
Ok, I will try it. Thank you for your response.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "skumra/robotic-grasping"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年7月14日(星期二) 下午2:04
收件人:&nbsp;"skumra/robotic-grasping"<robotic-grasping@noreply.github.com&gt;;
抄送:&nbsp;"戚宗城"<923746590@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [skumra/robotic-grasping] I can't reproduce the accuracy such as 97.8% and 96.8% as the paper presented. (#11)





 
You can normalize the Q output of the model if you want the output to be between 0 and 1.

**Comment by qzcmask** [comment]
Can you give me the normalize code to deal with Jacquard Dataset?

**Comment by skumra** [comment]
You can normalize the quality output by `q_img = q_img / np.max(q_img)` or clip it by `q_img = np.clip(q_img, a_min = 0, a_max = 1)`.

**Comment by skumra** [comment]
Also, did you try to reproduce the results on Cornell dataset using the latest release?

**Comment by qzcmask** [comment]
I have tryied that it reproduces a great result.


Nice work!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "skumra/robotic-grasping"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年7月16日(星期四) 上午7:50
收件人:&nbsp;"skumra/robotic-grasping"<robotic-grasping@noreply.github.com&gt;;
抄送:&nbsp;"戚宗城"<923746590@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [skumra/robotic-grasping] I can't reproduce the accuracy such as 97.8% and 96.8% as the paper presented. (#11)





 
Also, did you try to reproduce the results on Cornell dataset using the latest release?

**Comment by qzcmask** [comment]
After looking at the `train()` function, I realized that each epoch only trained a thousand samples. If I trained the Jacquard dataset, would the data that each epoch trained be the same (the first thousand samples)?
Did I get it wrong?

**Comment by skumra** [comment]
We use the `SubsetRandomSampler` to sample from the training dataset. It samples elements randomly from a given list of indices, without replacement.  The total number of samples per epoch is defined by `batch-size` x `batches-per-epoch`.
Note: The default parameters are tuned for Cornell dataset.

**Comment by skumra** [comment]
Closing the issue as new code and trained models were released and @qzcmask was able to reproduce the results.

**Comment by qzcmask** [comment]
Hi shumra. I have two questions about the experiment. Firstly, how do you split the Cornell Dataset in two way including object-wise and image-wise? Secondly, do you have augumented the Cornell Dataset(51k grasp examples) before training or augumenting it randomly during training? Hope your response.

**Comment by skumra** [comment]
The Cornell dataset if read sequentially is object-wise, and if you shuffle the dataset before splitting, it gets splits image-wise. This is done in the code using the `ds-shuffle` argument. 

The dataset is augmented randomly during training.

**Comment by kerwinwanga** [comment]
> 在查看了函数后，我意识到每个 epoch 只训练了 1000 个样本。如果我训练了 Jacquard 数据集，每个 epoch 训练的数据是否相同（前一千个样本）？我弄错了吗？`train()`

How is this problem solved in the jacquard dataset
